packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -31,6 +31,10 @@ import { GridColumnResizeParams } from './params/gridColumnResizeParams';
 import { GridColumnVisibilityChangeParams } from './params/gridColumnVisibilityChangeParams';
 import { GridClasses } from './gridClasses';
 
+export type MuiEvent<E> = (E | React.SyntheticEvent) & {
```

Comment:
@dtassone I added the `E` type variable so we can reuse it with different native events if necessary.

Reply:
Nice, but why not do an interface?

Reply:
Interfaces can't extend union types.

https://stackoverflow.com/questions/56085306/error-message-an-interface-can-only-extend-an-object-type-or-intersection-of-o

Reply:
because you shouldn't do a union type...
I would have done
```
interface MuiEvent<E> extends React.SyntheticEvent<E> {
...
}
```
But while typing, I noticed your `E | SyntheticEvent`. E is any, but yeah it could work anywhere 🤔

Reply:
Yeah, but this event when triggered by the `document` is a native event. The union needs to be there.